diff --git a/lib/utilsBundle.js b/lib/utilsBundle.js
index 188aa33465d8f91ca457237873952f966be4272e..b5a0062b2cecc9049e6542a331ca9000fd7a2d0c 100644
--- a/lib/utilsBundle.js
+++ b/lib/utilsBundle.js
@@ -58,7 +58,7 @@ const program = require("./utilsBundleImpl").program;
 const ProgramOption = require("./utilsBundleImpl").ProgramOption;
 const progress = require("./utilsBundleImpl").progress;
 const SocksProxyAgent = require("./utilsBundleImpl").SocksProxyAgent;
-const ws = require("./utilsBundleImpl").ws;
+const ws = 'Bun' in globalThis ? require('ws') : require('./utilsBundleImpl').ws;
 const wsServer = require("./utilsBundleImpl").wsServer;
 const wsReceiver = require("./utilsBundleImpl").wsReceiver;
 const wsSender = require("./utilsBundleImpl").wsSender;
